lib/tools/xctest-commands.js

```diff
@@ -1,24 +1,48 @@
 import { fixOutputToArray } from '../helpers';
+import _ from 'lodash';
 
 const xctestCommands = {};
+const IDB_ENV_PREFIX = 'IDB_';
+
+/**
+ * @typedef {Object} RunOptions
+ *
+ * @property {string} env Environment variables to be passed to the test runner
+ * @property {Array} args Arguments to be passed to the test runner
+ */
 
 /**
  * Runs an xcuitest on the device or simulator
  * @param {string} testRunnerBundleId The bundle id of the test runner
  * @param {string} appUnderTestBundleId The bundle id of the app under test
  * @param {string} xctestBundleId The bundle id of the xctest package
+ * @param {RunOptions} opts The envs and args to be passed to the xcuitest runner
  * @returns {SubProcess}
  */
-xctestCommands.runXCUITest = async function runXCUITest (testRunnerBundleId, appUnderTestBundleId, xctestBundleId) {
+xctestCommands.runXCUITest = async function runXCUITest (testRunnerBundleId, appUnderTestBundleId, xctestBundleId, opts = {}) {
   const uiTestProcess = this.createSubProcess([
     'xctest', 'run',
     'ui', xctestBundleId,
-    appUnderTestBundleId, testRunnerBundleId
-  ]);
+    appUnderTestBundleId, testRunnerBundleId,
+    ...(opts.args || [])
+  ], {
+    env: convertToIDBEnv(opts)
+  });
   await uiTestProcess.start(0);
   return uiTestProcess;
 };
 
+function convertToIDBEnv (opts) {
+  if (_.isEmpty(opts.env)) {
+    return null;
```

Comment:
null or undefined?

Reply:
I prefer using null. Is there a potential problem with that?

Reply:
for methods which return plain objects or arrays it might be, since JS will throw an exception if you try to deconstruct null, though everything will be fine if deconstructing an empty object/array

Reply:
regarding null/undefined - there is a diff when passing to JSON responses. 
`undefined` means the corresponding key will be removed from the resulting JSON, although `null` will be converted to JSON null and the key will be preserved. Then the behaviour depends on the actual processing logic on the other side

Reply:
I'm passing null since I expect that if you pass an empty object to child process, it doesn't pass the default envs to the process. Null or undefined should prevent that from happening

Reply:
Like here https://github.com/nodejs/node/blob/master/lib/child_process.js#L504. That is why we should pass `null` or `undefined`